apps/api/src/domain/models/types/HasEntityId.ts

```diff
@@ -0,0 +1,9 @@
+import { EntityId } from '../../types/ResourceId';
+
+export type HasEntityID<T = undefined> = T extends undefined
```

Comment:
Not void- in TypeScript, void is `undefined` when used as a return type from a function. But I can see why this seems strange. Look at one place where it is used. This trick is one I picked up from Giles. By defaulting the generic to undefined and then using the conditional type, you make providing the generic optional so that HasEntityId<MyType> = MyType & {id: EntityId}, while HasEntityId = {id: EntityId} (with nothing else).

That said, I ended up refactoring `HasEntityID` to be an interface instead, for other reasons.